### PR TITLE
cleanup focused on noise in logs

### PR DESCRIPTION
**Pull request by ethn**
(work in progress)

### CODE REVIEW

File: mod/metric_answer_lookup/lib/answer/answer_class_methods.rb [hunk 1]
```diff
@@ -74,10 +74,15 @@ def refresh ids=nil, *fields
     end
 
     def refresh_entry fields, ma_id
-      create_or_update ma_id, *fields
-    rescue => e
-      puts "failed to refresh metric answer: #{ma_id}"
-      puts e.message
+      if !Card[ma_id]
+        delete_answer_for_card_id ma_id
+      else
```
Comment (ethn) [inline mod/metric_answer_lookup/lib/answer/answer_class_methods.rb]: the old code was not handling the delete case, and this was leading to a lot of log noise, but also presumably a bunch of answers that no longer existed as cards.

File: mod/metrics/spec/set/type/metric_value_spec.rb [hunk 2]
```diff
@@ -97,8 +97,11 @@
       expect(new_answer.name)
         .to eq "#{metric.name}+#{company.name}+2015"
       new_name = "#{metric.name}+#{company.name}+2014"
-      new_answer.update_attributes! name: new_name
-      expect(new_answer.name).to eq(new_name)
+      newer_answer = new_answer.refresh true
+      # FIXME: this refresh prevents a log error (exception in integrate phase: undefined method `metric_card`)
+      # that error is happening because the director on new_answer is not getting cleared.
```
Comment (ethn) [inline mod/metrics/spec/set/type/metric_value_spec.rb]: See FIXME.  I went a little ways into looking into this and saw that ActManager.clear was getting called and that a director for the card in question was getting cleared, and yet it still had one at this point.  Not clear to me what's happening.

File: mod/metric_answer_lookup/lib/answer/answer_class_methods.rb (hunk 1 above)
Comment (xithan) [reply mod/metric_answer_lookup/lib/answer/answer_class_methods.rb]: The error message seems helpful to me (for example `delete_answer_for_card_id` will throw an exception if the entry doesn't exist). But would be better if the error is also send to Airbrake.

File: mod/metric_answer_lookup/lib/answer/answer_class_methods.rb [hunk 3]
```diff
@@ -74,10 +74,15 @@ def refresh ids=nil, *fields
     end
 
     def refresh_entry fields, ma_id
-      create_or_update ma_id, *fields
-    rescue => e
-      puts "failed to refresh metric answer: #{ma_id}"
-      puts e.message
+      if !Card[ma_id]
```
Comment (xithan) [inline mod/metric_answer_lookup/lib/answer/answer_class_methods.rb]: We don't need the card with all the set modules. `Card.exist?` ?

File: mod/metric_answer_lookup/lib/answer/answer_class_methods.rb (hunk 1 above)
Comment (xithan) [reply mod/metric_answer_lookup/lib/answer/answer_class_methods.rb]: I agree. My point was to raise a more meaningful error to make our work easier if a bug occurs. For example like this https://github.com/wikirate/wikirate/pull/48/files. That way you immediately see in what situation for which card the refresh failed. You don't have to set a breakpoint and try to reproduce the situation.

**Pull request merged**